core/src/main/scala/org/bykn/bosatsu/PackageCustoms.scala

```diff
@@ -224,4 +233,66 @@ object PackageCustoms {
       case Some(nel) => Validated.invalid(nel)
     }
   }
-}
+
+  private def noUselessBinds[A: HasRegion](pack: Package.Typed[A]): ValidatedNec[PackageError, Unit] = {
+    type Node = Either[pack.exports.type, Bindable]
+    implicit val ordNode: Ordering[Node] =
+      new Ordering[Node] {
+        def compare(x: Node, y: Node): Int =
+          x match {
+            case Right(bx) =>
+              y match {
+                case Left(_) => 1
+                case Right(by) =>
+                  Ordering[Identifier].compare(bx, by)
+              }
+            case Left(_) => 
+              y match {
+                case Left(_) => 0
+                case Right(_) => -1
+              }
+          }
+      }
+
+    val exports: Node = Left(pack.exports)
+    val roots: List[Node] =
+      (exports ::
+        pack.program.lets.collect { case (b, _, _) if b.asString.startsWith("_") => Right(b) } :::
```

Comment:
this should really be collecting all synthentic names and ignoring any synthetic name.

This probably should not be a root, just filtered from the errors. synthetic names are used to rewrite:
```
# top level
(a, b) = foo
```
into:
```
_a = foo
a = match _a:
  case (a, _): a
b = match _a:
  case (_, b): b
```
so, the only case where `_a` would be unused is for constructs like:
```
_ = foo(1)
```
where we intend this idiom to exist only to check that `foo(1)` type-checks, but can then be removed from runtime.